cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -0,0 +1,260 @@
+
+/**
+ * Status codes that indicate a Lead and Copper Rule violation.
+ * See https://www.epa.gov/sites/default/files/2019-07/documents/environments-and-contaminants-methods-drinking-water.pdf
+ * Page 4.
+ */
+const LEAD_AND_COPPER_VIOLATIONS = new Set(['57', '58', '59', '63', '65']);
+
+const COMPLIANCE_STATUS_MAP = new Map([
+  // Indicates the violation is not open but the system has not yet been
+  // designated as returned to compliance.
+  ['K', 'Known'],
+  ['O', 'Open'],
+  ['R', 'Return to compliance'],
+]);
+
+/**
+ * Inserts all rows into the violations table.
+ */
+async function insertRows(db: ConnectionPool, rows: ViolationsTableRow[]): Promise<any[]> {
+  return db.query(sql`INSERT INTO epa_violations (violation_id,
+                                                 pws_id,
+                                                 violation_code,
+                                                 compliance_status,
+                                                 start_date, geom)
+                      VALUES ${sql.join(
+                        rows.map((row: ViolationsTableRow) => {
+                          return sql`(${row.violation_id}, 
+                                      ${row.pws_id}, 
+                                      ${row.violation_code}, 
+                                      ${row.compliance_status}, 
+                                      ${row.start_date},
+                                      ${sql.__dangerous__rawValue(
+                                        `ST_AsText(ST_GeomFromGeoJSON('${row.geom}'))`,
+                                      )})`;
+                        }),
+                        ',',
+                      )};`);
+}
+
+/**
+ * Inserts all rows into the violations table.
+ */
+async function deleteRows(db: ConnectionPool): Promise<any[]> {
+  return db.query(sql`DELETE
+                      FROM epa_violations
+                      WHERE violation_id IS NOT NULL`);
+}
+
+/**
+ * Reads the S3 CSV file and the number of rows successfully written.
+ */
+function parseS3IntoViolationsTableRow(
+  s3Params: AWS.S3.GetObjectRequest,
+  db: ConnectionPool,
+  numberOfRowsToWrite = DEFAULT_NUMBER_ROWS_TO_INSERT,
+): Promise<number> {
+  return new Promise(function (resolve, reject) {
+    const batchSize = 10;
+    let numberRows = 0;
+    let results: ViolationsTableRow[] = [];
+
+    const fileStream = S3.getObject(s3Params).createReadStream();
+    let pipeline = chain([
+      fileStream,
+      Pick.withParser({ filter: 'features' }),
+      streamArray(),
+      new Batch({ batchSize: batchSize }),
+    ]);
+
+    pipeline
+      .on('data', async (batch: any[]) => {
+        if (numberRows < numberOfRowsToWrite) {
+          for (const data of batch) {
+            const value = data.value;
+            const properties = value.properties;
+
+            console.log(properties);
+            if (LEAD_AND_COPPER_VIOLATIONS.has(properties[VIOLATION_CODE])) {
```

Comment:
What does it mean if the `VIOLATION_CODE` is not in the `LEAD_AND_COPPER_VIOLATIONS` set?

Reply:
There's a lot more than lead and copper violations in this table: Go to page 4 of these docs outlining the breakdown https://www.epa.gov/sites/default/files/2019-07/documents/environments-and-contaminants-methods-drinking-water.pdf

Reply:
Done